src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Utilities/StringExtensions.cs

```diff
@@ -356,6 +358,19 @@ public static string TrimEnd(this string value, string trim = "")
             ? value[..^trim.Length]
             : value;
     }
+#endif
+
+    public static string TrimEndString(this string value, string suffix)
```

Comment:
Could you please elaborate what are the different behaviors?

In both cases, we can have the same name and move #if .. #elseif .. #endif inside the function, this way no need to have a new method name

Reply:
`TrimEnd` exists in net9.0, so we need to hide it when we build in net9.0.
And in net9.0 `TrimEnd("Part")` is equivalent to `TrimEnd(new char[] { 'P', 'a', 'r', 't' })` so `"MenuListPart"` would become `"MenuLis"` without the `t`.

Reply:
Seems a new overload has been added in .NET 9.0